client/checkout/classic/new-upe.js

```diff
@@ -0,0 +1,242 @@
+
+				return;
+			}
+		}
+
+		const elements = api.getStripe().elements( options );
+		gatewayUPEComponents[ paymentMethodType ].elements = elements;
+
+		const upeElement = elements.create( 'payment', {
+			...getUpeSettings(),
+			wallets: {
+				applePay: 'never',
+				googlePay: 'never',
+			},
+		} );
+
+		upeElement.mount( domElement );
+		gatewayUPEComponents[ paymentMethodType ].upeElement = upeElement;
+	}
+
+	let paymentMethodGenerated;
+
+	const handleCheckout = ( $form, paymentMethodType ) => {
+		if ( paymentMethodGenerated ) {
+			paymentMethodGenerated = null;
+			return;
+		}
+
+		blockUI( $form );
+		const elements = gatewayUPEComponents[ paymentMethodType ].elements;
+		elements.submit().then( ( result ) => {
+			if ( result.error ) {
+				showErrorCheckout( result.error.message );
+			}
+		} );
+		api.getStripe()
+			.createPaymentMethod( {
+				elements,
+				params: {
+					billing_details: {
+						name: $( '#billing_first_name' ).length
+							? (
+									$( '#billing_first_name' ).val() +
+									' ' +
+									$( '#billing_last_name' ).val()
+							  ).trim()
+							: undefined,
+						email: $( '#billing_email' ).val(),
+						phone: $( '#billing_phone' ).val(),
+						address: {
+							city: $( '#billing_city' ).val(),
+							country: $( '#billing_country' ).val(),
+							line1: $( '#billing_address_1' ).val(),
+							line2: $( '#billing_address_2' ).val(),
+							postal_code: $( '#billing_postcode' ).val(),
+							state: $( '#billing_state' ).val(),
+						},
+					},
+				},
+			} )
+			.then( function ( paymentMethodObject ) {
+				paymentMethodGenerated = true;
+
+				appendFingerprintInputToForm( $form, fingerprint );
+				appendPaymentMethodIdToForm(
+					$form,
+					paymentMethodObject.paymentMethod.id
+				);
+				$form.removeClass( 'processing' ).submit();
+			} )
+			.catch( ( error ) => {
+				$form.removeClass( 'processing' ).unblock();
+				showErrorCheckout( error.message );
+			} );
+		// Prevent form submission so that we can fire it once a payment method has been generated.
+		return false;
+	};
+
+	// Handle the checkout form when WooCommerce Payments is chosen.
+	const wcpayPaymentMethods = [
+		PAYMENT_METHOD_NAME_BANCONTACT,
+		PAYMENT_METHOD_NAME_BECS,
+		PAYMENT_METHOD_NAME_EPS,
+		PAYMENT_METHOD_NAME_GIROPAY,
+		PAYMENT_METHOD_NAME_IDEAL,
+		PAYMENT_METHOD_NAME_P24,
+		PAYMENT_METHOD_NAME_SEPA,
+		PAYMENT_METHOD_NAME_SOFORT,
+		PAYMENT_METHOD_NAME_CARD,
+	];
+	const checkoutEvents = wcpayPaymentMethods
+		.map( ( method ) => `checkout_place_order_${ method }` )
+		.join( ' ' );
+	$( 'form.checkout' ).on( checkoutEvents, function () {
+		const paymentMethodType = getSelectedUPEGatewayPaymentMethod();
+		return handleCheckout( $( this ), paymentMethodType );
+	} );
+
+	function appendPaymentMethodIdToForm( $form, paymentMethodId ) {
+		// Remove any existing payment method inputs.
+		$form.find( 'input[name="wcpay-payment-method"]' ).remove();
```

Comment:
Can there be multiple of these inputs on the page? Shouldn't this only get added once for the card payment method? Card payment method should be ever present, I believe, so I think we can rely on this input being present at least once.

Reply:
With the current implementation, this input is added only once for the payment gateway used for the purchase without the risk of being added more than once due to the point of time it gets added (after checkout processing starts).

This removal is rather a countermeasure from the weird cases when the input is already on the page. However, I can't think of any real use case when this could happen. I've removed this line in 02f4657e549bd178ffeb882a7a6ba09b3303a0c4. Please let me know if I somehow misunderstood your question or the purpose of this thread as such.